File: .NET/Microsoft.Recognizers.Text.DateTime/Utilities/DateTimeResolutionResult.cs
﻿// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

using System.Collections.Generic;

namespace Microsoft.Recognizers.Text.DateTime
{
    public class DateTimeResolutionResult
    {
        public DateTimeResolutionResult()
        {
            Success = false;
        }

        public bool Success { get; set; }

        public string Timex { get; set; }

        public bool IsLunar { get; set; }

        public string Mod { get; set; }

        public bool HasRangeChangingMod { get; set; } = false;

        public string Comment { get; set; }

        public Dictionary<string, string> FutureResolution { get; set; }

        public Dictionary<string, string> PastResolution { get; set; }

        public object FutureValue { get; set; }

        public object PastValue { get; set; }

        public List<object> SubDateTimeEntities { get; set; }

        public TimeZoneResolutionResult TimeZoneResolution { get; set; }

        public List<object> List { get; set; }
    }
}

File: Python/libraries/recognizers-date-time/recognizers_date_time/date_time/utilities/datetime_resolution_result.py
from typing import List, Dict
from recognizers_date_time.date_time.utilities import TimeZoneResolutionResult


class DateTimeResolutionResult:
    def __init__(self):
        self.success: bool = False
        self.timex: str = ''
        self.is_lunar: bool = False
        self.mod: str = ''
        self.has_range_changing_mod: bool = False
        self.comment: str = ''
        self.future_resolution: Dict[str, str] = dict()
        self.past_resolution: Dict[str, str] = dict()
        self.future_value: object = None
        self.past_value: object = None
        self.sub_date_time_entities: List[object] = list()
        self.timezone_resolution: TimeZoneResolutionResult()
        self.list: List[object] = list()
